test/unit/ssz_ex_test.exs

```diff
@@ -217,6 +218,102 @@ defmodule Unit.SSZExTest do
     assert root |> Base.encode16(case: :lower) == expected_value
   end
 
+  test "hash tree root of list of composite objects" do
+    ## list of containers
+    checkpoint = %Checkpoint{
+      epoch: 12_345,
+      root: Base.decode16!("0100000000000000000000000000000000000000000000000000000000000001")
+    }
+
+    module = SszEx.get_module(checkpoint)
+
+    list = [checkpoint, checkpoint]
+    schema = {:list, module, 8}
+    SszEx.hash_tree_root!(list, schema)
+
+    ## list of lists
+    list1 = Stream.cycle([65_535]) |> Enum.take(316)
+    list2 = Stream.cycle([65_530]) |> Enum.take(316)
+    list = [list1, list2]
+    schema = {:list, {:list, {:int, 16}, 1024}, 1024}
+    SszEx.hash_tree_root!(list, schema)
+
+    ## list of list of lists
+    list1 = Stream.cycle([65_535]) |> Enum.take(316)
+    list2 = Stream.cycle([65_530]) |> Enum.take(316)
+    list3 = [list1, list2]
+    list4 = [list1, list2]
+    list = [list3, list4]
+    schema = {:list, {:list, {:list, {:int, 16}, 1024}, 1024}, 128}
+    SszEx.hash_tree_root!(list, schema)
+
+    ## list of list of vectors
+    list1 = Stream.cycle([65_535]) |> Enum.take(316)
+    list2 = Stream.cycle([65_530]) |> Enum.take(316)
+    list3 = [list1, list2]
+    list4 = [list1, list2]
+    list = [list3, list4]
+    schema = {:list, {:list, {:vector, {:int, 16}, 2}, 1024}, 136}
+    SszEx.hash_tree_root!(list, schema)
+
+    ## list of vector of lists
+    list1 = Stream.cycle([65_535]) |> Enum.take(316)
+    list2 = Stream.cycle([65_530]) |> Enum.take(316)
+    list3 = [list1, list2]
+    list4 = [list1, list2]
+    list = [list3, list4]
+    schema = {:list, {:vector, {:list, {:int, 16}, 1024}, 2}, 32}
+    SszEx.hash_tree_root!(list, schema)
+
+    ## list of vector of vector
+    vector1 = Stream.cycle([65_535]) |> Enum.take(316)
+    vector2 = Stream.cycle([65_530]) |> Enum.take(316)
+    vector3 = [vector1, vector2]
+    vector4 = [vector1, vector2]
+    list = [vector3, vector4]
+    schema = {:list, {:vector, {:vector, {:int, 16}, 316}, 2}, 32}
+    SszEx.hash_tree_root!(list, schema)
+  end
+
+  test "hash tree root of vector of composite objects" do
+    ## list of containers
+    checkpoint = %Checkpoint{
+      epoch: 12_345,
+      root: Base.decode16!("0100000000000000000000000000000000000000000000000000000000000001")
+    }
+
+    module = SszEx.get_module(checkpoint)
```

Comment:
This should be the same as using `Checkpoint`. Also, you can just do `%module{} = checkpoint` here